File: frontend/src/roadmap/ruby/final_challenge/task.rb
# TODO: Create Task class

class Task
  # Your code here
  
  def initialize(title, description = "", due_date = nil, priority = "medium", status = "pending")
    # Your code here
  end
  
  def to_s
    # Your code here - return string representation
  end
  
  def completed?
    # Your code here
  end
  
  def mark_complete!
    # Your code here
  end
  
  def overdue?
    # Your code here - check if due date has passed
  end
end

File: frontend/src/roadmap/python/final_challenge/task.py
# TODO: Create Task class

class Task:
    def __init__(self, title, description="", due_date=None, priority="medium", status="pending"):
        # Your code here
        pass
    
    def __str__(self):
        # Your code here - return string representation
        pass
    
    def to_dict(self):
        # Convert task to dictionary for JSON serialization
        pass
    
    @classmethod
    def from_dict(cls, data):
        # Create Task instance from dictionary
        pass
    
    def mark_complete(self):
        # Your code here
        pass
    
    def update_priority(self, new_priority):
        # Your code here
        pass
